scripts/zones/Aht_Urhgan_Whitegate/npcs/Maudaal.lua

```diff
@@ -3,16 +3,33 @@
 --  NPC: Maudaal
 -- Standard Info NPC
 -----------------------------------
+local ID = require("scripts/zones/Aht_Urhgan_Whitegate/IDs")
+require("scripts/globals/npc_util")
 
 function onTrade(player,npc,trade)
 end
 
 function onTrigger(player,npc)
-    player:startEvent(240)
+    if player:getQuestStatus(AHT_URHGAN,tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER) == QUEST_AVAILABLE then
+        player:startEvent(889)
+    elseif player:getQuestStatus(AHT_URHGAN,tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER) == QUEST_ACCEPTED and player:getCharVar("princeandhopper") == 6 then
+        player:startEvent(890)
+    else
+        player:startEvent(240)
+    end
 end
-
 function onEventUpdate(player,csid,option)
 end
 
 function onEventFinish(player,csid,option)
+    if (csid == 889 and option == 1) then
+        return
+    elseif (csid ==889 and option ==2) then
+        player:addQuest(AHT_URHGAN,tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER)
+        player:setCharVar("princeandhopper",1)
+    elseif (csid ==890) then
+        npcUtil.giveItem(player, 16270)
+        player:setCharVar("princeandhopper",0)
+        player:completeQuest(AHT_URHGAN,tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER)
+    end
```

Comment:
```lua
    elseif (csid ==890) then
        if npcUtil.giveItem(player, 16270) then
            player:setCharVar("princeandhopper",0)
            player:completeQuest(AHT_URHGAN,tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER)
        end
    end
```
Doing it like this will ensure they get their item before marking as complete. If it fails to give them the item they have to make room and watch the cs again.

Reply:
Or:
```
elseif csid == 890 then
    npcUtil.completeQuest(
        player,
        AHT_URHGAN,
        tpz.quest.id.ahtUrhgan.THE_PRINCE_AND_THE_HOPPER,
        {item = 16270, var = "princeandhopper"}
    )
end
```

npcUtil.completeQuest handles:
awarding items, keyitems, gil, XP, title, and fame
setting quest-related variables to 0
completing the quest

Reply:
I guess the best question is what is going to be the standard moving forward?

Reply:
That's much better, I didn't even realize we had that.